clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -983,10 +983,14 @@ void CodeGenFunction::StartFunction(GlobalDecl GD, QualType RetTy,
 
   if (getLangOpts().SYCLIsDevice && D) {
     if (const auto *A = D->getAttr<SYCLIntelLoopFuseAttr>()) {
-      Expr *E = A->getValue();
+      const Expr *Arg = A->getValue();
+      assert(Arg && "Got an unexpected null argument");
+      const auto *CE = dyn_cast<ConstantExpr>(Arg);
+      assert(CE && "Not an integer constant expression");
```

Comment:
Rather than do `dyn_cast<>` followed by an assertion, it's better to use `cast<>` because `cast<>` asserts that the returned value will not be null. (Basically, you use `dyn_cast<>` when the cast operation can fail and you use `cast<>` when the cast operation is expected to never fail.)

Reply:
Done